azkaban-common/src/test/java/azkaban/utils/FileIOUtilsTest.java

```diff
@@ -38,16 +39,47 @@
   public TemporaryFolder temp = new TemporaryFolder();
   private File sourceDir, destDir, baseDir;
 
+  /**
+   * Create a very big dir which would cause linux shell command to hard link the dir to
+   * exceed the allowed limit.
+   */
+  private void createBigDir(final String path) throws IOException {
+    final String verylongprefix =
+        "123123123123123123123123123123123113123111111111111111111111111111111111111111111111111111"
+            + "111111111111111111111111111111111111111111111111111111111111111111111111111111111111"
+            + "11231312312312312312313121111111111111111111111111111111111111111111111111";
+
+    for (int i = 1; i <= 400; i++) {
+      final File tmpDir = new File(path + "/" + verylongprefix + "dir" + i);
+      tmpDir.mkdir();
+      for (int j = 1; j <= 100; j++) {
+        final File tmp = new File(tmpDir.getAbsolutePath() + "/" + j);
+        tmp.createNewFile();
+      }
+    }
+  }
+
   @Before
   public void setUp() throws Exception {
     // setup base dir
+
     this.baseDir = this.temp.newFolder("base");
     final File file1 = new File(this.baseDir.getAbsolutePath() + "/a.out");
     final File file2 = new File(this.baseDir.getAbsolutePath() + "/testdir");
     final File file3 = new File(file2.getAbsolutePath() + "/b.out");
+    final File file4 = new File(file2.getAbsolutePath() + "/testdir");
+    final File file5 = new File(file2.getAbsolutePath() + "/c.out");
+    final File file6 = new File(file4.getAbsolutePath() + "/c1.out");
+    final File file7 = new File(file4.getAbsolutePath() + "/c2.out");
+
     file1.createNewFile();
     file2.mkdir();
     file3.createNewFile();
+    file4.mkdir();
+    file5.createNewFile();
+    file6.createNewFile();
+    file7.createNewFile();
+    createBigDir(this.baseDir.getAbsolutePath());
```

Comment:
Thanks for improving the tests.

I'd suggest a separate test for the special long argument case. This will keep the tests more focused and intention clearer.

Reply:
updated